### PR TITLE
dumbpad refactor - adding support for various PCB revisions

### PR DESCRIPTION
**Pull request by imchipwood**
## Description

Refactored keyboards/dumbpad to include support for various PCB revisions. Added readme.mk and info.json files for all revisions. ~~Added users/imchipwood folder~~

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fauxpark** [comment]
Please split the keyboard/keymap/user changes into separate PRs.

**Comment by imchipwood** [comment]
Can you clarify what you mean by requesting separate PRs?

The change to the structure of the dumbpad folder is large so I'm not sure exactly what you're asking for. 

- Separate PRs for: user code, default keyboard code, and the 'imchipwood' keymaps in dumbpad?
- Separate PRs for: original dumbpad code refactored (v0x), adding new dumbpad versions (v0x_*, v1x*), and user code?

Thanks!

**Comment by fauxpark** [comment]
Basically, do the refactoring in one PR and add your keymap & user code in another.

**Comment by imchipwood** [comment]
OK, user code and all user keymaps have been removed from dumbpad. Will submit second PR for those after this is merged.

File: keyboards/dumbpad/rules.mk [hunk 1]
```diff
@@ -35,3 +35,5 @@ HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs
 ENCODER_ENABLE = yes
 MOUSEKEY_ENABLE = yes
 KEY_LOCK_ENABLE = yes
+
+DEFAULT_FOLDER = dumbpad/v1x
```
Comment (noroadsleft) [inline keyboards/dumbpad/rules.mk:39]: The `v0x` version seems to be the device supported by the code this PR deprecates.

Behaviour should be maintained for existing users, with the goal to remove this rule (and thus require a version specified when using `make`/`qmk compile`) down the road.

```suggestion
DEFAULT_FOLDER = dumbpad/v0x
```

Comment (imchipwood) [reply keyboards/dumbpad/rules.mk:39]: Yes, v0x was the original design - it's technically archived in the board repo but I understand leaving it as the default for anyone who is still using v0x.

File: keyboards/dumbpad/info.json [hunk 2]
```diff
@@ -1,30 +1,6 @@
 {
```
Comment (fauxpark) [inline keyboards/dumbpad/info.json:1]: This file can be deleted.

File: keyboards/dumbpad/v0x_dualencoder/config.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define DEVICE_VER      0x0006
+#define DESCRIPTION     4x4 macro/numpad with two rotary encoders
```
Comment (fauxpark) [inline keyboards/dumbpad/v0x_dualencoder/config.h:23]:
```suggestion
```

File: keyboards/dumbpad/v0x_right/config.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+/* USB Device descriptor parameter */
+#define DEVICE_VER      0x0007
+#define DESCRIPTION     4x4 macro/numpad with rotary encoder on the right side
```
Comment (fauxpark) [inline keyboards/dumbpad/v0x_right/config.h:21]:
```suggestion
```

File: keyboards/dumbpad/v0x_right/info.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
+    "keyboard_name": "dumbpad/v0x_right",
+    "keyboard_folder": "dumbpad/v0x_right",
+    "width": 5,
+    "height": 4,
+    "layouts": {
+        "LAYOUT": {
+            "key_count": 17,
```
Comment (fauxpark) [inline keyboards/dumbpad/v0x_right/info.json:8]:
```suggestion
```

File: keyboards/dumbpad/v1x/config.h [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define DEVICE_VER      0x0010
+#define DESCRIPTION     4x4 macro/numpad with rotary encoder
```
Comment (fauxpark) [inline keyboards/dumbpad/v1x/config.h:23]:
```suggestion
```

File: keyboards/dumbpad/v1x/info.json [hunk 7]
```diff
@@ -0,0 +1,17 @@
+{
+    "keyboard_name": "dumbpad/v1x",
+    "keyboard_folder": "dumbpad/v1x",
+    "width": 5,
+    "height": 4,
+    "layouts": {
+        "LAYOUT": {
+            "key_count": 17,
```
Comment (fauxpark) [inline keyboards/dumbpad/v1x/info.json:8]:
```suggestion
```

File: keyboards/dumbpad/v1x_dualencoder/config.h [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define DEVICE_VER      0x0010
+#define DESCRIPTION     4x4 macro/numpad with two rotary encoders
```
Comment (fauxpark) [inline keyboards/dumbpad/v1x_dualencoder/config.h:23]:
```suggestion
```

File: keyboards/dumbpad/v1x_dualencoder/info.json [hunk 9]
```diff
@@ -0,0 +1,17 @@
+{
+    "keyboard_name": "dumbpad/v1x_dualencoder",
+    "keyboard_folder": "dumbpad/v1x_dualencoder",
+    "width": 4,
+    "height": 4,
+    "layouts": {
+        "LAYOUT": {
+            "key_count": 16,
```
Comment (fauxpark) [inline keyboards/dumbpad/v1x_dualencoder/info.json:8]:
```suggestion
```

File: keyboards/dumbpad/v1x_right/config.h [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define DEVICE_VER      0x0010
+#define DESCRIPTION     4x4 macro/numpad with rotary encoder on the right side
```
Comment (fauxpark) [inline keyboards/dumbpad/v1x_right/config.h:23]:
```suggestion
```

File: keyboards/dumbpad/v1x_right/info.json [hunk 11]
```diff
@@ -0,0 +1,17 @@
+{
+    "keyboard_name": "dumbpad/v1x_right",
+    "keyboard_folder": "dumbpad/v1x_right",
+    "width": 5,
+    "height": 4,
+    "layouts": {
+        "LAYOUT": {
+            "key_count": 17,
```
Comment (fauxpark) [inline keyboards/dumbpad/v1x_right/info.json:8]:
```suggestion
```

**Comment by imchipwood** [comment]
Hey - can someone please reopen this pull request? It appears I don't have permission to reopen it myself.

I have implemented the suggestions in my branch and would like to move forward! 

Thank you!

**Comment by drashna** [comment]
Looks like there is a merge conflict that needs resolution, now. Sorry

**Comment by imchipwood** [comment]
Merge conflict resolved.

**Comment by imchipwood** [comment]
@fauxpark @noroadsleft I believe all of the changes you requested have been implemented - when you have some time, please take a look once again! Thank you!

**Comment by imchipwood** [comment]
Hey QMK folks - just hoping to get a review on this some time soon, thanks!

File: keyboards/dumbpad/v0x/rules.mk [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (noroadsleft) [inline keyboards/dumbpad/v0x/rules.mk:11]:
```suggestion
```

File: keyboards/dumbpad/v0x/rules.mk [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
```
Comment (noroadsleft) [inline keyboards/dumbpad/v0x/rules.mk:21]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/dumbpad/v0x_dualencoder/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+                // other layers - =/+ (quals/plus) (CW) and -/_ (minus/underscore) (CCW)
+                if (clockwise) {
+                    tap_code(KC_EQL);
+                } else {
+                    tap_code(KC_MINS);
+                }
+                break;
+        }
+    } else if (index == 1) {
+        switch (biton32(layer_state)) {
```
Comment (noroadsleft) [inline keyboards/dumbpad/v0x_dualencoder/keymaps/default/keymap.c:114]:
```suggestion
        switch (get_highest_layer(layer_state)) {
```

File: keyboards/dumbpad/v0x_dualencoder/rules.mk [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (noroadsleft) [inline keyboards/dumbpad/v0x_dualencoder/rules.mk:11]:
```suggestion
```

File: keyboards/dumbpad/v0x_dualencoder/rules.mk [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
```
Comment (noroadsleft) [inline keyboards/dumbpad/v0x_dualencoder/rules.mk:21]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/dumbpad/v0x_dualencoder/templates/keymap.c [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+            default:
+                if (clockwise) {
+                    tap_code(KC_EQL);
+                } else {
+                    tap_code(KC_MINS);
+                }
+                break;
+        }
+    } else if (index == 1) {
+        switch (biton32(layer_state)) {
```
Comment (noroadsleft) [inline keyboards/dumbpad/v0x_dualencoder/templates/keymap.c:25]:
```suggestion
        switch (get_highest_layer(layer_state)) {
```

File: keyboards/dumbpad/v0x_right/rules.mk [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (noroadsleft) [inline keyboards/dumbpad/v0x_right/rules.mk:11]:
```suggestion
```

File: keyboards/dumbpad/v0x_right/rules.mk [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
```
Comment (noroadsleft) [inline keyboards/dumbpad/v0x_right/rules.mk:21]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/dumbpad/v1x/rules.mk [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (noroadsleft) [inline keyboards/dumbpad/v1x/rules.mk:11]:
```suggestion
```

File: keyboards/dumbpad/v1x/rules.mk [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
```
Comment (noroadsleft) [inline keyboards/dumbpad/v1x/rules.mk:21]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/dumbpad/v1x_dualencoder/keymaps/default/keymap.c [hunk 22]
```diff
@@ -0,0 +1,134 @@
...
+                // other layers - =/+ (quals/plus) (CW) and -/_ (minus/underscore) (CCW)
+                if (clockwise) {
+                    tap_code(KC_EQL);
+                } else {
+                    tap_code(KC_MINS);
+                }
+                break;
+        }
+    } else if (index == 1) {
+        switch (biton32(layer_state)) {
```
Comment (noroadsleft) [inline keyboards/dumbpad/v1x_dualencoder/keymaps/default/keymap.c:114]:
```suggestion
        switch (get_highest_layer(layer_state)) {
```

File: keyboards/dumbpad/v1x_dualencoder/readme.md [hunk 23]
```diff
@@ -0,0 +1,87 @@
+# dumbpad v1.x dual-encoder
+
+![dumbpad](v1x_dualencoder.png)
```
Comment (noroadsleft) [inline keyboards/dumbpad/v1x_dualencoder/readme.md:3]: The photo here should be uploaded to some place like Imgur and linked here, and then `keyboards/dumbpad/v1x_dualencoder/v1x_dualencoder.png` should be deleted.

From the [Docs](https://docs.qmk.fm/#/hardware_keyboard_guidelines?id=imagehardware-files "QMK Keyboard Guidelines"):

> In an effort to keep the repo size down we’re no longer accepting binary files of any format, with few exceptions. Hosting them elsewhere (such as https://imgur.com) and linking them in the readme.md is preferred.

File: keyboards/dumbpad/v1x_dualencoder/rules.mk [hunk 24]
```diff
@@ -0,0 +1,24 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (noroadsleft) [inline keyboards/dumbpad/v1x_dualencoder/rules.mk:11]:
```suggestion
```

File: keyboards/dumbpad/v1x_dualencoder/rules.mk [hunk 25]
```diff
@@ -0,0 +1,24 @@
...
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
```
Comment (noroadsleft) [inline keyboards/dumbpad/v1x_dualencoder/rules.mk:21]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/dumbpad/v1x_dualencoder/templates/keymap.c [hunk 26]
```diff
@@ -0,0 +1,43 @@
...
+            default:
+                if (clockwise) {
+                    tap_code(KC_EQL);
+                } else {
+                    tap_code(KC_MINS);
+                }
+                break;
+        }
+    } else if (index == 1) {
+        switch (biton32(layer_state)) {
```
Comment (noroadsleft) [inline keyboards/dumbpad/v1x_dualencoder/templates/keymap.c:25]:
```suggestion
        switch (get_highest_layer(layer_state)) {
```

File: keyboards/dumbpad/v1x_right/rules.mk [hunk 27]
```diff
@@ -0,0 +1,24 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (noroadsleft) [inline keyboards/dumbpad/v1x_right/rules.mk:11]:
```suggestion
```

File: keyboards/dumbpad/v1x_right/rules.mk [hunk 28]
```diff
@@ -0,0 +1,24 @@
...
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
```
Comment (noroadsleft) [inline keyboards/dumbpad/v1x_right/rules.mk:21]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/dumbpad/rules.mk [hunk 29]
```diff
@@ -1,35 +1 @@
...
-RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
-MIDI_ENABLE = no            # MIDI support
-UNICODE_ENABLE = no         # Unicode
-BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
-AUDIO_ENABLE = no           # Audio output on port C6
-
-ENCODER_ENABLE = yes
-MOUSEKEY_ENABLE = yes
-KEY_LOCK_ENABLE = yes
+DEFAULT_FOLDER = dumbpad/v1x
```
Comment (noroadsleft) [inline keyboards/dumbpad/rules.mk:1]: As previously discussed, this should be set to maintain behaviour for existing users.

```suggestion
DEFAULT_FOLDER = dumbpad/v0x
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
All of these are minor cleanup tasks to make this codebase more consistent with the rest of QMK.

**Comment by noroadsleft** [comment]
Still some items outstanding from my previous review:

- https://github.com/qmk/qmk_firmware/pull/9259#discussion_r587714219
- https://github.com/qmk/qmk_firmware/pull/9259#discussion_r587717811
- https://github.com/qmk/qmk_firmware/pull/9259#discussion_r587718674
- https://github.com/qmk/qmk_firmware/pull/9259#discussion_r587719266
- https://github.com/qmk/qmk_firmware/pull/9259#discussion_r587719576
- https://github.com/qmk/qmk_firmware/pull/9259#discussion_r587720308
- https://github.com/qmk/qmk_firmware/pull/9259#discussion_r587720550
- https://github.com/qmk/qmk_firmware/pull/9259#discussion_r587721190

File: keyboards/dumbpad/v1x_dualencoder/readme.md (hunk 23 above)
Comment (imchipwood) [reply keyboards/dumbpad/v1x_dualencoder/readme.md]: Done!

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
:tada:

**Pull request merged**